File: src/programmers/level2/기능_개발.java
package programmers.level2;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class 기능_개발 {
    public static void main(String[] args) {
        int[] progresses = {95, 90, 99, 99, 80, 99};
        int[] speeds = {1, 1, 1, 1, 1, 1};
        solution(progresses, speeds);
    }

    public static int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> daysQueue = new LinkedList<>();
        List<Integer> list = new ArrayList<>();

        for (int i = 0; i < progresses.length; i++) {
            double quotient = (100 - progresses[i]) / (double) speeds[i];
            int day = (int) Math.ceil(quotient);

            if (!daysQueue.isEmpty() && daysQueue.peek() < day) {
                list.add(daysQueue.size());
                daysQueue.clear();
                daysQueue.add(day);
            } else
                daysQueue.add(day);
        }

        list.add(daysQueue.size());

        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let stack = [];
    let cnt = 1;
    
    // 작업 완료까지 걸리는 일수 계산
    for(let i = progresses.length - 1; i >= 0; i--){
        stack.push(Math.ceil((100 - progresses[i]) / speeds[i]))
    }
    
    let target = stack.pop();
    while(stack.length > 0){
        let lastStackItem = stack.pop();
        
        if(target < lastStackItem){
            answer.push(cnt);
            cnt = 1; // 초기화
            target = lastStackItem;
        } else cnt++;
    }
    
    answer.push(cnt);
    return answer;
}